source/extensions/common/wasm/stats_handler.cc

```diff
@@ -96,6 +96,16 @@ void StatsHandler::onEvent(WasmEvent event) const {
   case WasmEvent::VmReloadFailure:
     wasm_stats_.vm_reload_failure_.inc();
     break;
+  // VM failure events - increment the universal vm_failed counter
+  case WasmEvent::UnableToCreateVm:
+  case WasmEvent::UnableToCloneVm:
+  case WasmEvent::MissingFunction:
+  case WasmEvent::UnableToInitializeCode:
+  case WasmEvent::StartFailed:
+  case WasmEvent::ConfigureFailed:
```

Comment:
Are these events emitted somewhere? I don't see where these events are passed-through.

Reply:
the events are published from here: https://github.com/envoyproxy/envoy/blob/f384ab2b3e3aa0564ef25f57dc2ed8ad61eaf0cb/source/extensions/common/wasm/wasm.cc#L279

And newly added 
  stats_handler_->onEvent(WasmEvent::RuntimeError);

Reply:
Thank you for reviewing 🙇

Reply:
I see. Could you add a test for one of the other failure events, besides the reloadable runtime error. That would demonstrate that the stats handler is being invoked.